### PR TITLE
Feature/reordable links

### LINKED ISSUE #820: Reorderable links in Quicklinks
**Is your feature request related to a problem? Please describe.**
It would be nice to have a reorderable quicklinks page so all links can be reorder, remove or added. For this to work, we would need to maintain 1 list of all links (never changes) and one list of currently displayed links (part of the complete list). An index of each link must be saved to be able to track where the item is located.

We would also need to save theses infos in localstorage.

**Acceptance criteria**
- [ ] On page open, order the list based on the order saved;
- [ ] On long press of a quicklink, activate the reactivate feature and show small 'x' to delete a link;
- [ ] Add a way to create a link that is disappear (can't create 2 identical links too);
- [ ] On a press anywhere else without links, we should save once again the quicklinks with the order in localstorage;
- [ ] Make sure to have an initializer to insert default layout when none is found;
- [ ] Tests 🧪  

**Describe alternatives you've considered**
N/A

**Additional context**
This package could be easily used 😄 https://pub.dev/packages/reorderable_grid_view

Also, it's to note that with the new horairÉTS link, this feature is more urgent now since we don't necessarily want to schedule genrator at the same time.

### PR DESCRIPTION
**Pull request by JonathanDuvalV**
### ⁉️ Related Issue
Closes #820 

## 📖 Description

This PR makes the quicklinks in the "ETS" tab reordable by holding for 1 second on any of the WebLinkCard. The QuickLinks can also be deleted or added back

Here is a short demo:

https://github.com/ApplETS/Notre-Dame/assets/45638733/3d36d5eb-9add-4c14-9723-460d3c02bc41

### 🧪 How Has This Been Tested?
Tests were added for the new viewmodel and repository for QuickLinks. Other quicklink related tests were modified to fit the new structure.

### ☑️ Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.
- [x] If needed, I added analytics.
- [x] Make sure to add either one of the following labels: `version: Major`,`version: Minor` or `version: Patch`. 
- [x] Make sure golden files changes were reviewed and approved.

### CODE REVIEW

**Comment by JonathanDuvalV** [comment]
J'ai pas pensé en faisant la tâche 😅, mais on voudrait quoi comme couleur pour la bulle du "x" en light theme? 

image

File: lib/core/managers/quick_link_repository.dart [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import 'dart:convert';
```
Comment (camillebrulotte) [inline lib/core/managers/quick_link_repository.dart:1]: Ohohoh police des imports ici

File: lib/core/viewmodels/quick_links_viewmodel.dart [hunk 2]
```diff
@@ -1,16 +1,87 @@
 // FLUTTER / DART / THIRD-PARTIES
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:notredame/core/managers/quick_link_repository.dart';
```
Comment (camillebrulotte) [inline lib/core/viewmodels/quick_links_viewmodel.dart:3]: 🚨🚨

File: lib/locator.dart [hunk 3]
```diff
@@ -2,6 +2,7 @@
 import 'package:flutter_secure_storage/flutter_secure_storage.dart';
 import 'package:get_it/get_it.dart';
 import 'package:logger/logger.dart';
+import 'package:notredame/core/managers/quick_link_repository.dart';
```
Comment (camillebrulotte) [inline lib/locator.dart:5]: 🚨🚨

File: lib/ui/views/quick_links_view.dart [hunk 4]
```diff
@@ -1,8 +1,13 @@
 // FLUTTER / DART / THIRD-PARTIES
 import 'package:flutter/material.dart';
+import 'package:notredame/ui/utils/app_theme.dart';
```
Comment (camillebrulotte) [inline lib/ui/views/quick_links_view.dart:3]: 🚨🚨

File: test/helpers.dart [hunk 5]
```diff
@@ -6,6 +6,7 @@ import 'package:flutter_secure_storage/flutter_secure_storage.dart';
 import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:logger/logger.dart';
+import 'package:notredame/core/managers/quick_link_repository.dart';
```
Comment (camillebrulotte) [inline test/helpers.dart:9]: 🚨🚨

File: test/managers/quick_link_repository_test.dart [hunk 6]
```diff
@@ -0,0 +1,98 @@
+// FLUTTER / DART / THIRD-PARTIES
+import 'dart:convert';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:notredame/core/managers/quick_link_repository.dart';
```
Comment (camillebrulotte) [inline test/managers/quick_link_repository_test.dart:6]: 🚨🚨

File: test/managers/quick_link_repository_test.dart [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+// SERVICES / MANAGER
+import 'package:notredame/core/managers/cache_manager.dart';
+import 'package:notredame/core/models/quick_link.dart';
```
Comment (camillebrulotte) [inline test/managers/quick_link_repository_test.dart:10]: 🚨🚨

File: test/ui/views/quick_links_view_test.dart [hunk 8]
```diff
@@ -7,6 +7,7 @@ import 'package:flutter_gen/gen_l10n/app_localizations.dart';
 
 // CONSTANTS
 import 'package:notredame/core/constants/quick_links.dart';
+import 'package:notredame/core/managers/quick_link_repository.dart';
```
Comment (camillebrulotte) [inline test/ui/views/quick_links_view_test.dart:10]: 🚨🚨

File: test/ui/views/quick_links_view_test.dart [hunk 9]
```diff
@@ -19,13 +20,16 @@ import 'package:notredame/ui/views/quick_links_view.dart';
 import 'package:notredame/ui/widgets/web_link_card.dart';
 
 import '../../helpers.dart';
+import '../../mock/managers/quick_links_repository_mock.dart';
```
Comment (camillebrulotte) [inline test/ui/views/quick_links_view_test.dart:23]: 🚨🚨

File: test/viewmodels/quick_links_viewmodel_test.dart [hunk 10]
```diff
@@ -0,0 +1,123 @@
+// FLUTTER / DART / THIRD-PARTIES
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:notredame/core/constants/quick_links.dart';
```
Comment (camillebrulotte) [inline test/viewmodels/quick_links_viewmodel_test.dart:4]: 🚨🚨

**Comment by camillebrulotte** [comment]
La couleur c'est fine ! C'est comme les icones dans la bar de nav ! Good job à part pour les imports !! 🚨 Je rigole

**Review by XavierPaquet-Rapold — DISMISSED** [review]
LGTM 💯

**Review by camillebrulotte — APPROVED** [review]
LGTM !

**Pull request merged**